### ISSUE TITLE
No CLI for listing disabled plugins

### ISSUE THREAD

**Issue by vidartf** [jupyterlab/jupyterlab#6751]
**Describe the bug**
While `jupyter labextension list` will output information about disabled *extensions*, it does not provide any information about disabled *plugins*.

**To Reproduce**

```
jupyter labextension disable @jupyterlab/application-extension:layout
jupyter labextension list
```

**Expected behavior**
I would expect disabled plugins to be listed as well. Either separately, or possibly indented beneath the owning extension for non-core extensions.

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser: N/A
- JupyterLab: 1.0.0

**Comment by jtpio** [comment]
Something like this?

```
JupyterLab v1.1.0a1
No installed extensions

Disabled core extensions:
    @jupyterlab/application-extension:layout
```

Even though it says "extensions", it's probably better than not showing them at all.

**Comment by vidartf** [comment]
Yes, something like that, but if we put both extensions and plugins under the same heading we might as well change the heading to "plugins/extensions" :)

I would probably not list disabled plugins if the extension itself is uninstalled/disabled.

**Comment by jtpio** [comment]
Also, running a `jupyter labextension disable` command to disable a plugin might sound a bit odd. But at the same time we probably don't want to introduce a new `jupyter labplugin`.

**Comment by sheromon** [comment]
I was confused by the behavior of `jupyter labextenion list` recently. I found that some JupyterLab extension features were separated into multiple plugins by the developers to allow fine-grained enabling/disabling of features, like in #10086, which was the solution to the request in #9770.

Similarly, I found that I was able to disable the [RetroLab interface switcher](https://github.com/jupyterlab/retrolab/blob/main/packages/lab-extension/src/index.ts#L59) by disabling just that plugin. Very convenient, and it shows up as disabled when I run `jupyter labextension list`, for example 
```
/opt/homebrew/Caskroom/miniforge/base/envs/jupyterlab/share/jupyter/labextensions
        jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
        @retrolab/lab-extension v0.3.21 enabled OK

Disabled extensions:
    @retrolab/lab-extension:interface-switcher
```

Based on these examples, I expected similar behavior with my own custom JupyterLab extension. I wrote it with two plugins, and I found that I was able to individually enable/disable each, but the status is not reflected in `jupyter labextension list`. 

Any thoughts on why the different behavior for disabling a single RetroLab plugin vs. disabling a single plugin from my custom extension? (I can post example code if desired.)

**Comment by krassowski** [comment]
@sheromon previously a similar issue was reported on Jupyter Discourse [here](https://discourse.jupyter.org/t/pre-built-extensions-and-labextension-disable-not-working/15318/9?u=krassowski) and turned out to be a typo. If it is not a typo, a fully reproducible minimal example would help a lot!

If it is a typo, it means that there is a problem in the `disable` command: it does not warn if attempting to disable a plugin which does not exist and we should fix it.

**Comment by krassowski** [comment]
As for the original issue it seems that it was fixed over two years ago as the plugins are now included (and extensions denoted with suffix "all plugins", as of https://github.com/jupyterlab/jupyterlab/pull/9492). We might want to finish it of by changing the section title to `Disabled extensions/plugins`.

**Comment by sheromon** [comment]
Thanks for the quick response, @krassowski. I actually mentioned this earlier on Jupyter discourse, and you let me know I can open an issue, haha. But I searched and found this one that sounded like the same problem! Anyway, I took a little time to make my extension super simple, and now I have two plugins that just log messages to the console. I went off of the tutorial material, so I used cookie-cutter to set up my project initially and then mainly edited the `index.ts` file. 

My super-simple `index.ts` has this code.
```
import {
  JupyterFrontEnd,
  JupyterFrontEndPlugin
} from '@jupyterlab/application';


const log1Plugin: JupyterFrontEndPlugin<void> = {
  id: '@jupyterlab-cdanner/lab-extension:log1-plugin',
  autoStart: true,
  activate: (app: JupyterFrontEnd) => {
    console.log('log1 plugin is running');
  }
};

const log2Plugin: JupyterFrontEndPlugin<void> = {
  id: '@jupyterlab-cdanner/lab-extension:log2-plugin',
  autoStart: true,
  activate: (app: JupyterFrontEnd) => {
    console.log('log2 plugin is running');
  }
};

const plugins: JupyterFrontEndPlugin<any>[] = [
  log1Plugin,
  log2Plugin,
];

export default plugins;
```

I built the pip-installable package by running `python -m build`, and I installed it in a conda virtual environment by running `pip install` on the wheel file that came out. By default, both plugins are enabled, so I see two messages in my browser console when I start JupyterLab, as expected.

If I run `jupyter labextension disable @jupyterlab-cdanner/lab-extension:log2-plugin` and then restart JupyterLab, only the first message gets logged and not the second, as expected. But my output from `jupyter labextension list` is still the same.
```
JupyterLab v3.4.6
/opt/homebrew/Caskroom/miniforge/base/envs/jupyter3/share/jupyter/labextensions
        jupyterlab-cdanner v0.1.0 enabled OK (python, jupyterlab_cdanner)
        jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
```

In other words, I know the enable/disable of individual plugins is working because I can see the behavior change, but the listed status doesn't match. When I initially observed this behavior, I had many other Python packages installed, but my most recent testing, I had only JupyterLab (3.4.6) and my custom extension installed in my Python 3.9 conda environment.

**Comment by sheromon** [comment]
@krassowski, I've got it working now, finally! I didn't catch what OPs problem was (mismatch in `package.json`) in the discussion you linked earlier until today. I had seen other people describe problems enabling or disabling their extensions/plugins because they made a typo, and the name they typed didn't match the plugin name, but I knew what I had match the plugin name because I copy-pasted it straight from the code.

My conclusion is that my problem stems from some rules or things that need to match that I wasn't aware of, one being the name in the `package.json` file (I think?), and another being... I'm not sure on this either, but it seems like the name of the plugin has to match the directory structure or lack thereof to be recognized by `jupyter labextension disable`. My interpretation is that once I had my plugins named in a way that was consistent with the rules, now the enabled/disabled status shows correctly.
